source/application/tasks/commonjs/index.js

```diff
@@ -1,217 +1,103 @@
-				let environmentConfig = patternItem.environments[resultEnvironmentName].manifest;
-				metaData.results[resultEnvironmentName] = {};
-
-				for (let resultName of Object.keys(resultEnvironment)) {
-					let result = resultEnvironment[resultName];
-					metaData.results[resultEnvironmentName][resultName] = {};
-
-					let variants = [
-						{ 'name': 'demo', 'buffer': result.demoBuffer },
-						{ 'name': '', 'buffer': result.buffer }
-					];
-					variants = variants.filter((item) => item.buffer.length > 0);
-
-					// Write all variants into snippets
-					for (let variant of variants) {
-						let fragments = [
-							resultName.toLowerCase(),
-							!environmentConfig.formats || environmentConfig.formats.includes(result.in) ? resultEnvironmentName : '',
-							variants.length > 1 ? variant.name : ''
-						];
-
-						fragments = fragments.filter((item) => item);
-
-						let resultFileBaseName = `${fragments.join('-')}.${result.out}`;
-						let resultFile = resolve(patternSnippetsDirectory, resultFileBaseName);
-						await qfs.write(resultFile, variant.buffer);
-						metaData.results[resultEnvironmentName][resultName][variant.name || 'library'] = qfs.relativeFromDirectory(patternResultDirectory, resultFile);
-					}
-
-					// Write main variant into pattern build folder, render html into layout
-					let mainBuffer = result.buffer;
-					let mainName = resultName.toLowerCase();
-
-					// TODO: resolve this
-					if (mainName === 'markup') {
-						continue;
-					}
-
-					// TODO: resolve this
-					mainName = mainName === 'documentation' ? mainName : resultEnvironmentName;
-
-					let mainFileBaseName = `${mainName}.${result.out}`;
-					let mainFile = resolve(patternResultDirectory, mainFileBaseName);
-					await qfs.write(mainFile, mainBuffer);
-				}
-			}
-
-			let commonjsPkg = {
-				name: pkg.name,
-				version,
-				dependencies: {
-					react: pkg.dependencies.react || require(require.resolve('patternplate/node_modules/patternplate-server/package.json')).dependencies.react
-				}
-			};
-			await qfs.write(resolve(commonjsRoot, 'package.json'), JSON.stringify(commonjsPkg, null, '  '));
-		}
+	if (application.cache) {
+		application.cache.config.static = true;
 	}
 
-	// Build environment output
-	if (commonjsConfig.tasks.bundles) {
-		let environments = await qfs.listTree(resolve(patternRoot, '@environments'));
+	for (const patternItem of patternList) {
+		const resultEnvironment = patternItem.results.index;
 
-		environments = environments
-			.filter((item) => basename(item) === 'pattern.json')
-			.map((item) => dirname(item));
+		for (const resultName of Object.keys(resultEnvironment)) {
+			const result = resultEnvironment[resultName];
+			const resultPath = join(...[commonjsRoot, ...resolveResultPath(patternItem.id, resultName, result.out)]);
+			const resultDirectory = dirname(resultPath);
 
-		if (environments.length === 0) {
-			environments = ['index'];
+			await qfs.makeTree(resultDirectory);
```

Comment:
I would prefer if you use mkdirp here instead of qfs and below the fs.writeFile API.
qfs should become deprecated in patternplate.